docker/service.go

```diff
@@ -300,14 +278,28 @@ func (s *Service) up(ctx context.Context, imageName string, create bool, options
 		containers = []*Container{c}
 	}
 
-	return s.eachContainer(ctx, containers, func(c *Container) error {
-		if create {
-			if err := s.recreateIfNeeded(ctx, imageName, c, options.NoRecreate, options.ForceRecreate); err != nil {
+	hasBeenRecreate := false
+
+	return s.watchContainers(ctx, containers, func(c *Container) error {
```

Comment:
yeah there is too much `return` here. I didn't want to change too luch the current code base, but definitely room for improvement (making the inline func real function, less return, etc..)